hack/verify-go-src.sh

```diff
@@ -15,80 +15,29 @@
 
 set -euo pipefail
 
```

Comment:
Can you add some documentation describing what this script is doing? It's not immediately obvious IMO from the code, and I think a quick summary will make maintenance easier.

Reply:
Will do.